### PR TITLE
Partially added Python type hints and enabled mypy in CI

### LINKED ISSUE #483: Add Python type hints + enable mypy in CI
Type hints are useful both when reading/understanding the source code, as well as useful to catch bugs upfront during development.

Objective of this issue is to add type hints `webviz-subsurface`. The corresponding PR in `webviz-config` is https://github.com/equinor/webviz-config/pull/195, and the [CI line there is this one](https://github.com/equinor/webviz-config/blob/8428a68b80e9a310fcef35604cbd2dfd984b4e32/.github/workflows/webviz-config.yml#L73).

If during the type hint work suggestions for repo restructuring are found, these could either be documented as new issues (large restructuring) or done at the same time (small restructuring). See also #464.

_More info regarding mypy: https://github.com/python/mypy_

### PR DESCRIPTION
**Pull request by rubenthoms**
This closes #483. Probably requires a new issue addressing type hinting of the remaining files.

### CODE REVIEW

File: .pylintrc
Comment (anders-kiaer) [inline .pylintrc:27]: Missing newline.

File: tests/conftest.py
Comment (anders-kiaer) [inline tests/conftest.py:3]:
```suggestion
from typing import Any
import pathlib
```
We have so far used PEP8 group import (https://www.python.org/dev/peps/pep-0008/#imports).
1. Standard libraries.
2. External libraries.
3. Same Package submodules.

File: tests/integration_tests/test_aggregated_data_example.py
Comment (anders-kiaer) [inline tests/integration_tests/test_aggregated_data_example.py:5]:
```suggestion
import sys
import subprocess  # nosec
from pathlib import Path

from dash.testing.composite import DashComposite
```
Relevant in some of the other files as well.

**Review by anders-kiaer — COMMENTED** [review]
Awesome work 🎉 Could not find any big things to comment on (minor comments only).

Line 18 in the GitHub workflow can perhaps be reactivated now (since it is a PR)? And we can generalize the workflow in a separate PR to also work without a PR.

@asnyv 's PR will fix `webviz-config`, we do a release, and can then enable 93+94 as well.

🚀

**Review by HansKallekleiv — COMMENTED** [review]
Great work! LGTM

**Comment by anders-kiaer** [comment]
@rubenthoms @sigurdp Is CI in this PR happy if we revert back to running `webviz docs` and temp. install `webviz-config` from `master`? Then we will follow up by releasing a bug fix of `webviz-config` and we take this one in :tada:

**Comment by sigurdp** [comment]
Just tested running `webviz docs` locally with `webviz-config` from `master` installed and it is running fine here.

**Comment by rubenthoms** [comment]
> Just tested running `webviz docs` locally with `webviz-config` from `master` installed and it is running fine here.

Same here. Ready to be taken in :) I'd wait until this is merged in before starting with the new PVT plot features in order to have the type hints included.

File: .github/workflows/subsurface.yml
Comment (anders-kiaer) [inline .github/workflows/subsurface.yml:18]:
```suggestion
    if: github.event_name != 'push' || github.ref == 'refs/heads/master' || contains(github.event.head_commit.message, '[deploy test]')
```

File: .github/workflows/subsurface.yml
Comment (anders-kiaer) [inline .github/workflows/subsurface.yml:94]:
```suggestion
        webviz docs --portable ./docs_build --skip-open
```

**Review by anders-kiaer — COMMENTED** [review]
🎉

**Review by anders-kiaer — APPROVED** [review]
Great work @rubenthoms and @sigurdp :tada: Feel free to merge when :green_circle:

**Pull request merged**
